wsr-tic-tac-toe/src/__tests__/game-board.test.tsx

```diff
@@ -0,0 +1,226 @@
+        const messageBoxDriver = MessageBoxFunctionalLayoutTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_MESSAGE_BOX,
+        });
+        await messageBoxDriver.clickOnConfirmationButton();
+        const squareCount = document.evaluate(
+            'count(//div[contains(@class,"square")])',
+            document,
+            null,
+            XPathResult.ANY_TYPE,
+            null
+        ).numberValue;
+        expect(squareCount).toEqual(4);
+    });
+
+    it('should change turn order after clicking a square', async () => {
+        const createGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_BUTTON,
+        });
+
+        await createGameButtonDriver.click();
+        const inputDriver = InputWithLabelTestkit({ wrapper: wrapper, dataHook: BOARD_SIZE_INPUT });
+
+        await inputDriver.enterText('2');
+        const messageBoxDriver = MessageBoxFunctionalLayoutTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_MESSAGE_BOX,
+        });
+        await messageBoxDriver.clickOnConfirmationButton();
+        const square0 = document.getElementById('0');
+        square0?.click()
+        const turnOrderHeadingDriver = HeadingTestkit({ wrapper, dataHook: TURN_ORDER_HEADING });
+        expect(await turnOrderHeadingDriver.exists()).toBeTruthy();
+        expect(await turnOrderHeadingDriver.getText()).toEqual("Player2's turn");
+    });
+
+    it('should reset all squares after clicking reset game button', async () => {
+        const createGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_BUTTON,
+        });
+
+        await createGameButtonDriver.click();
+        const inputDriver = InputWithLabelTestkit({ wrapper: wrapper, dataHook: BOARD_SIZE_INPUT });
+
+        await inputDriver.enterText('2');
+        const messageBoxDriver = MessageBoxFunctionalLayoutTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_MESSAGE_BOX,
+        });
+        await messageBoxDriver.clickOnConfirmationButton();
+        const square0 = document.getElementById('0');
+        // click a square, so the state would change from initial state
+        square0?.click()
+        const resetGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: RESET_GAME_BUTTON,
+        });
+        // click reset button to reset game state
+        await resetGameButtonDriver.click()
+
+        // only gets squares which are unclicked
+        const squareCount = document.evaluate(
+            'count(//div[contains(@class,"square")])',
+            document,
+            null,
+            XPathResult.ANY_TYPE,
+            null
+        ).numberValue;
+        const turnOrderHeadingDriver = HeadingTestkit({ wrapper, dataHook: TURN_ORDER_HEADING });
+        expect(await turnOrderHeadingDriver.getText() === "Player1's turn" && squareCount === 4).toBeTruthy();
```

Comment:
don't "and" expectations
write separate expectations for each thing you are testing. If you "and" them, you won't see which part of "and" failed